### ISSUE TITLE
Clarity for the NOTE included in the "programmatically determined link context" definition

### ISSUE THREAD

**Issue by giacomo-petri** [w3c/wcag#3809]
The definition of [programmatically determined link context](https://www.w3.org/TR/WCAG22/#dfn-programmatically-determined-link-context) includes a note stating:

> Since screen readers interpret punctuation, they can also provide the context from the current sentence, when the focus is on a link in that sentence.

During discussions within the ACT community group about related "Programmatically Determined Link Context" rules (https://github.com/act-rules/act-rules.github.io/issues/2176#issuecomment-2089981563), we've identified inconsistencies in AT/UA support regarding the context provided by sentences.

As evidenced by examples outlined in https://github.com/act-rules/act-rules.github.io/issues/2176#issuecomment-2089981563, support for sentence context, which seems to be lacking in NVDA, appears deficient and even when present, it's not consistently reliable.

Should we consider addressing this aspect by modifying the NOTE?

**Comment by giacomo-petri** [comment]
Hi @mraccess77,

> If the screen reader can read back the paragraph - and a paragraph includes the current sentence I wonder if this is already included by meeting that. Is there a situation where a sentence would be outside of a paragraph?

In the [linked examples](https://github.com/act-rules/act-rules.github.io/issues/2176#issuecomment-2089981563) you can gain further insight into the issue at hand. Consider this scenario:
```
<p>
    <span><a href="link1">Read more</a> about this.</span>
    <span>"Other content here".</span>
    <span><a href="link2">Read more</a> about that.</span>
    <span>"Other content here".</span>
    <span><a href="link3">Read more</a> about this and that.</span>
</p>
```
or simply
```
<p><a href="link1">Read more</a> about this.</span><a href="link2">Read more</a> about that.</p>
```

The problem is that by simply reading the paragraph might not enable you to differentiate between the links. However, the ability of UA/AT to recognize punctuation to read sentences - as emphasized in the note within the definition - can make a significant difference in discerning between the links, but it seems not true at the moment.

**Comment by mbgower** [comment]
I concur with @mraccess77. 
There is _nothing_ that can be rendered programmatically that can be derived from that demonstration paragraph, as presented to users. Forcing authors to add new context programmatically for screen reader users that is not present for other users tends to be a bad idea. The solution should be to provide better context overall for everyone, and ensure that everyone can access that context equitably.

It's important to emphasize that there _is_ no sentence level programmatic indicator (in HTML). It is derived from the syntax of the text. We may want to tweak the note @giacomo-petri references at the top of this issue to make that clearer.

I've also made some [comments](https://github.com/act-rules/act-rules.github.io/issues/2176#issuecomment-3036634441) on the linked examples in the referenced issue.

**Comment by mbgower** [comment]
Actually, thinking about this further, I think the clarification also needs to happen in [Link Purpose (In Context](https://www.w3.org/WAI/WCAG21/Understanding/link-purpose-in-context), in regard to the normative wording:
> or from the link text together with its [programmatically determined link context](https://www.w3.org/WAI/WCAG21/Understanding/link-purpose-in-context#dfn-programmatically-determined-link-context),

The AAA version seems fine since it is explicitly about the link text alone.

Also, note that _both_ versions include the language "except where the purpose of the link would be [ambiguous to users in general](https://www.w3.org/WAI/WCAG21/Understanding/link-purpose-link-only#dfn-ambiguous-to-users-in-general)", which supports the sentiment echoed by @mraccess77 earlier.

Suggested steps for TF:
- A new section that covers concepts of punctuation and syntax (as opposed to programmatic indicators) could be added to one (or both?) Understanding documents.
- Consider tweaking the language in the note of the definition to add some more context

**Comment by giacomo-petri** [comment]
Sorry, I missed the earlier comments.

in 
`<p><a href="link1">Read more</a> about topic 1.</span><a href="link2">Read more</a> about topic 2.</p>`

> It seems like that multiple link issue is a challenge for all users and not just screen reader users.

I disagree with your statement that this example is ambiguous in general.

As presented, visually I can clearly associate the first "Read more" with the first sentence (about topic 1) and the second "Read more" with the second sentence (about topic 2). However, if I land on the second "Read more" using a screen reader and I ask the AT to read the context, which is presented as:

> Read more about topic 1. Read more about topic 2.

I'm not able to understand if the read more I'm going to activate is the 1st one or the 2nd one. 

So I would argue to your comments saying that the intent of the rule, which is to disambiguate the link thanks to its content, doesn't work very well in this case if the puntuation is not considered by ATs.
